pkg/controller/nodenetworkconfigurationpolicy/policyconditions/conditions_test.go

```diff
@@ -51,44 +51,67 @@ func p(conditionsSetter func(*nmstatev1alpha1.ConditionList, string), message st
 	}
 }
 
-func newNode(idx int, conditions []corev1.NodeCondition) corev1.Node {
-	nodeName := fmt.Sprintf("node%d", idx)
+func nodeName(idx int) string {
+	return fmt.Sprintf("node%d", idx)
+}
+
+func newNode(idx int) corev1.Node {
+	nodeName := nodeName(idx)
 	node := corev1.Node{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: nodeName,
 			Labels: map[string]string{
 				"kubernetes.io/hostname": nodeName,
 			},
 		},
-		Status: corev1.NodeStatus{
-			Conditions: conditions,
-		},
 	}
 	return node
 }
 
-func nodeReady() []corev1.NodeCondition {
-	return []corev1.NodeCondition{
-		corev1.NodeCondition{
-			Type:   corev1.NodeReady,
-			Status: corev1.ConditionTrue,
+func newPodAtNode(idx int) corev1.Pod {
+	pod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("foo-%d", idx),
+			Namespace: "bar",
+			Labels: map[string]string{
+				"app": "foo-bar-zar",
+			},
+		},
+		Spec: corev1.PodSpec{
+			NodeName: nodeName(idx),
 		},
 	}
+	return pod
 }
 
-func nodeNotReady() []corev1.NodeCondition {
-	return []corev1.NodeCondition{
-		corev1.NodeCondition{
-			Type:   corev1.NodeReady,
-			Status: corev1.ConditionFalse,
+func newNmstatePodAtNode(idx int) corev1.Pod {
```

Comment:
We can pass name and app to newPodAtNode and reuse it for both. Let's not get flooded by helper functions.

Reply:
Well I am going to add another helper function :-P, but going to make a generic one that compose the Pod struct.